### PR TITLE
Improve pymysql types

### PR DESCRIPTION
**Pull request by vrih**
Added show_warnings to connections. No return type specified as this
requires tracing back through missing return types in previous
author's work. This is out of scope for this pull request.

Added SSDictCursor, SSCursor and DictCursorMixin to complete the
Cursors definition.

### CODE REVIEW

File: third_party/2and3/pymysql/cursors.pyi [hunk 1]
```diff
@@ -22,12 +22,24 @@ class Cursor:
...
     def __iter__(self): ...
 
 class DictCursor(Cursor):
     def fetchone(self) -> Optional[Dict[str, Any]]: ...
-    def fetchmany(self, size=None) -> Optional[Tuple[Dict[str, Any], ...]]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> Optional[Tuple[Dict[str, Any], ...]]: ...
     def fetchall(self) -> Optional[Tuple[Dict[str, Any], ...]]: ...
+
+class DictCursorMixin:
+    dict_type = ...  # type: dict
```
Comment (JelleZijlstra) [inline third_party/2and3/pymysql/cursors.pyi]: Don't have time to check right now, but should this be `Type[Dict]`?

Also, it's preferred to use PEP 526 syntax (`dict_type: Type[Dict] = ...`).

File: third_party/2and3/pymysql/cursors.pyi [hunk 2]
```diff
@@ -22,12 +22,24 @@ class Cursor:
...
+
+class SSCursor(Cursor):
+    def fetchall(self) -> Optional[Tuple[Gen, ...]]: ...
+    def fetchall_unbuffered(self) -> Iterator[Tuple[Gen, ...]]: ...
+    def __iter__(self) -> Iterator[Tuple[Gen, ...]]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> List[Gen]: ...
+    def scroll(self, value, mode: str = ...): ...
```
Comment (JelleZijlstra) [inline third_party/2and3/pymysql/cursors.pyi]: Missing a type for `value` and a return type.

Comment (vrih) [reply third_party/2and3/pymysql/cursors.pyi]: Pushed a commit to correct this. I copied the types from the scroll in the pre-existing class for this. I've corrected there as well.

File: third_party/2and3/pymysql/cursors.pyi (hunk 1 above)
Comment (vrih) [reply third_party/2and3/pymysql/cursors.pyi]: I'm changing this to `# type: Any` for now. From the pymysql code it needs something dict like, but not sure what that means specifically for types.

File: third_party/2and3/pymysql/cursors.pyi [hunk 3]
```diff
@@ -22,12 +22,24 @@ class Cursor:
...
     def fetchone(self) -> Optional[Dict[str, Any]]: ...
-    def fetchmany(self, size=None) -> Optional[Tuple[Dict[str, Any], ...]]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> Optional[Tuple[Dict[str, Any], ...]]: ...
     def fetchall(self) -> Optional[Tuple[Dict[str, Any], ...]]: ...
+
+class DictCursorMixin:
+    dict_type = ...  # type: Any
+
+class SSCursor(Cursor):
+    def fetchall(self) -> Optional[Tuple[Gen, ...]]: ...
```
Comment (JelleZijlstra) [inline third_party/2and3/pymysql/cursors.pyi]: Doesn't this return a list? https://github.com/PyMySQL/PyMySQL/blob/master/pymysql/cursors.py#L458

File: third_party/2and3/pymysql/cursors.pyi [hunk 4]
```diff
@@ -22,12 +22,24 @@ class Cursor:
...
+    def fetchall_unbuffered(self) -> Iterator[Tuple[Gen, ...]]: ...
+    def __iter__(self) -> Iterator[Tuple[Gen, ...]]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> List[Gen]: ...
+    def scroll(self, value: int, mode: str = ...): ...
```
Comment (JelleZijlstra) [inline third_party/2and3/pymysql/cursors.pyi]: Missing return type (`None`).

File: third_party/2and3/pymysql/cursors.pyi (hunk 3 above)
Comment (JelleZijlstra) [reply third_party/2and3/pymysql/cursors.pyi]: I think you should give the correct type (`List[Gen]`) here, and `# type: ignore` the error. We should give the types as they actually are, even if they break LSP.

**Pull request merged**